### PR TITLE
Feature/rh sr hits

### PR DESCRIPTION
**Pull request by rhowell42**
Added variable filling block to fill newly added SRHits and SRSpacePoints. It is dependent on this PR being accepted https://github.com/SBNSoftware/sbnanaobj/pull/72 

Fill block is inside a switch to fill or not fill those new variables, default and value in this PR is set to false; so this will have no impact on computing resources for production

### CODE REVIEW

**Review by etyley — APPROVED** [review]
A few minor comments, but otherwise looks good

File: sbncode/CAFMaker/CAFMakerParams.h [hunk 1]
```diff
@@ -133,12 +133,24 @@ namespace caf
...
+      false // default is false; set to true if you want SRHits filled
+    };
+
     Atom<string> HitLabel {
       Name("HitLabel"),
       Comment("Base label of the TPC Hit producer."),
       "gaushit"
     };
 
+    Atom<string> ClusterPFParticleLabel {
```
[the remark this answers is not in the document]
Comment (rhowell42) [reply sbncode/CAFMaker/CAFMakerParams.h:148]: Ah yeah, that was another vestible label. I'll remove that

**Comment by rhowell42** [comment]
I added a setDefault method to FillReco.cxx at @brucehoward-physics  recommendation in my most recent commit to this branch

File: sbncode/CAFMaker/FillReco.cxx [hunk 2]
```diff
@@ -713,6 +713,31 @@ namespace caf
     }
   }
 
+  void FillHitVars(const recob::Hit& hit,
+                   unsigned producer,
+                   const recob::SpacePoint& spacepoint,
+                   const recob::PFParticle& particle,
+                   caf::SRHit& srhit,
+                   bool allowEmpty)
```
Comment (brucehoward-physics) [inline sbncode/CAFMaker/FillReco.cxx:721]: I'm not a reviewer on this PR but does this parameter do anything?

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 3]
```diff
@@ -1358,6 +1378,31 @@ void CAFMaker::produce(art::Event& evt) noexcept {
...
+        }
+        if (!thisParticle.empty() && !thisPoint.empty()) {
+          assert(thisParticle.size() == 1);
+          assert(thisPoint.size() == 1);
+          rec.reco.nhit++;
+          rec.reco.hit.push_back(SRHit());
+
+          FillHitVars(thisHit, producer, *thisPoint[0], *thisParticle[0], rec.reco.hit.back());
+          recslc.reco.hit.push_back(rec.reco.hit.back());
+          recslc.reco.nhit = recslc.reco.hit.size();
```
Comment (brucehoward-physics) [inline sbncode/CAFMaker/CAFMaker_module.cc:1402]: Possibly I'm completely overlooking something here as I was quickly looking but would `nhit =` be better done outside the loop?

**Comment by rhowell42** [comment]
I resolved all the comments here and in my pull request in sbnanaobj. The comments in this PR aren't completely resolved to me, which you can see in my response to Bruce above. Otherwise, if by validation run you mean if I successfully built and created created CAFs with these PRs, then yes I have. See /icarus/app/users/rhowell/SRHits/test.{flat}.caf.root

**Pull request merged**